modules/topic/index.js

```diff
@@ -0,0 +1,79 @@
+let bot;
+let chunks = null;
+let numChunksAllowed = 16;
+const chunkFileName = 'chunks.json';
+
+module.exports.init = function (b) {
+  bot = b;
+  bot.readDataFile(chunkFileName, (err, jsonData) => {
+    if (err) {
+      bot.log.info('initializing chunks');
+      chunks = {};
+    } else {
+      try {
+        chunks = JSON.parse(jsonData);
+      } catch (ex) {
+        bot.log.warn('Corrupted chunks file! Resetting dict...');
+        chunks = {};
+      }
+    }
+  });
+}
+
+  let store = (chunk, user) => {
+    chunks.chunk = user;
+    bot.writeDataFile(chunkFileName, JSON.stringify(chunks), (err) => {
+      if (err) bot.log.error(`Error writing chunks file: ${err}`);
+    });
+  };
+
+module.exports.commands = {
+  topic(a, parts, b, c, user, channel) {
+    let msg;
+    chan = bot.client.chanData(channel);
+    switch(parts[0]) {
```

Comment:
Note that you could also make it something like:

```
module.exports.commands {
  topic: {
    help(_, parts, reply, _, ...) {
      ...
    }
  }
}
```

The sugar for defining commands as object keys does allow nesting.

You can define a function named `_default` under `topic` if you e.g. want `!topic` with no additional parts to print out help as well.